### ISSUE TITLE
using GCODE for shutdown in .gcode files fails

### ISSUE THREAD

**Issue by ruedli** [jneilliii/OctoPrint-Tasmota#198]
Very strange, must be something simple...

I configured a tasmota plug, the check shows it "works", se e my config and check:

![image](https://github.com/jneilliii/OctoPrint-Tasmota/assets/5008440/f1f56b8e-ac28-4c30-8b1e-1ddd07be441e)

When I use the icon (the bolt), it turns from green to red, shutsdown octoprint and the relay turns of the power after the configured delay.

When I enter the command in a gcode file -or- enter it as an ending gcode in otorpint, it is NOT executed.

My gcode:
![image](https://github.com/jneilliii/OctoPrint-Tasmota/assets/5008440/2acec210-3727-4833-a94d-fecdd6871b02)

My end-script:
![image](https://github.com/jneilliii/OctoPrint-Tasmota/assets/5008440/596bfeef-8b0d-45a4-b5b3-99a176c09389)

This is shown in the command log:
![image](https://github.com/jneilliii/OctoPrint-Tasmota/assets/5008440/7209c18f-835a-41cc-b371-3ddc72c58bc8)

Now comes the strange thing: When I use the SEND command button (for sending gcode from the web interface)... it works!
The log shows the same:
![image](https://github.com/jneilliii/OctoPrint-Tasmota/assets/5008440/38f1dbf0-e1c6-40e3-9d9a-e2341a85997e)

So: bolt turns red, it does a shutsdown and the relay turns off.
![image](https://github.com/jneilliii/OctoPrint-Tasmota/assets/5008440/187ccfd1-35cd-41ac-a8d4-1acaf219b8a6)

Is my gcode wrong? It is the same as what I pasted and sent.

**Comment by ruedli** [comment]
I have the impression, the command simply does not "reach" the plugin when played through the GCODE printing stream. Perhaps because it is distorted?

**Comment by jneilliii** [comment]
try increasing your gcode off delay. I think what you are running into is a race condition where the plugin is processing the off command but the printer is still in a printing state and not finished yet. set that value to like 15-30 seconds and see if that plays nicely.

**Comment by ruedli** [comment]
That did the trick. I also found that my raspberry Pi needed substantial more time to fully come to a stop. The octopi server looses the connection rapidly, but the shutdown takes longer. I found these delays working for my setup.

(Pi powered from the PSU of the printer with a 5V / 5A stepdown convertor, and Tasmota relay in the mains cable to the printer)

![image](https://github.com/jneilliii/OctoPrint-Tasmota/assets/5008440/9b9e7794-f3bd-4a39-8c4b-0ca0c33bac7c)
